source/monitoring.txt

```diff
@@ -0,0 +1,401 @@
+
+Once you have started your server, open JConsole in your terminal using the
+following command:
+
+.. code-block:: shell
+
+   jconsole
+
+Once JConsole is open, perform the following actions in the GUI:
+
+#. Select the process running the preceding example code.
+#. Press :guilabel:`Insecure Connection` in the warning dialog box.
+#. Click on the :guilabel:`MBeans` tab.
+#. Inspect your connection pool events under the ``"org.mongodb.driver"`` domain.
+
+When you no longer want to inspect your connection pools in JConsole, do the
+following:
+
+#. Exit JConsole by closing the JConsole window
+#. Stop the program running by the preceding code snippet
+
+For more information on JMX and JConsole, see the following resources from
+Oracle:
+
+- `JConsole Documentation <https://www.oracle.com/technical-resources/articles/java/jconsole.html>`__
+- `Monitoring and Management Guide <https://docs.oracle.com/en/java/javase/16/management/monitoring-and-management-using-jmx-technology.html>`__
+
+For more information on the ``JMXConnectionPoolListener`` class, see
+the API documentation for
+`JMXConnectionPoolListener <{+core-api+}/com/mongodb/management/JMXConnectionPoolListener.html>`__.
+
+Include the Driver in Your Distributed Tracing System
+-----------------------------------------------------
+
+If you use a **distributed tracing system**, you can include event data from the
+driver. A distributed tracing system is an application that
+tracks requests as they propagate throughout different services in a
+service-oriented architecture.
+
+If you use the driver in a `Spring Cloud <https://spring.io/projects/spring-cloud>`__
+application, use
+`Spring Cloud Sleuth <https://spring.io/projects/spring-cloud-sleuth>`__ to
+include MongoDB event data in the
+`Zipkin <https://zipkin.io/>`__ distributed tracing system.
+
+If you do not use Spring Cloud or want to include driver event data in a distributed
+tracing system other than Zipkin, you must write a command event listener that
+manages `spans
+<https://docs.spring.io/spring-cloud-sleuth/docs/current-SNAPSHOT/reference/html/getting-started.html#getting-started-terminology>`__
+for your desired distributed tracing system. To see an implementation of such a
+listener, see the Java source code for the :github:`TraceMongoCommandListener <spring-cloud/spring-cloud-sleuth/blob/eb01d5952b2e736970efbe3fc7e9784d119eeae9/spring-cloud-sleuth-instrumentation/src/main/java/org/springframework/cloud/sleuth/instrument/mongodb/TraceMongoCommandListener.java>`
+class in the Spring Cloud Sleuth source code.
+
+To learn more about Spring Cloud Sleuth, see
+`Getting Started <https://docs.spring.io/spring-cloud-sleuth/docs/current-SNAPSHOT/reference/html/getting-started.html>`__
+in the Spring Cloud Sleuth documentation.
+
+To view a detailed description of a distributed tracing system, see
+`Dapper <https://research.google/pubs/pub36356/>`__ from Google Research.
+
+.. tip::
```

Comment:
S: since you have this admonition, I think you can remove the inline links (Spring Cloud, spans, Zipkin, etc) as well as the two paragraphs above ("To learn more about Spring Cloud Sleuth... To view a detailed description of a distribute tracing system ...")